data/transactions/logic/assembler.go

```diff
@@ -1125,6 +1130,59 @@ func typeTxField(pgm ProgramKnowledge, args []string) (StackTypes, StackTypes) {
 	return StackTypes{fs.ftype}, nil
 }
 
+func typeStore(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes) {
+	if len(args) == 0 {
+		return nil, nil
+	}
+	n, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return nil, nil
+	}
+	scratchIndex := int(n)
+	top := len(pgm.stack) - 1
+	if top >= 0 && scratchIndex < 256 && scratchIndex >= 0 {
+		pgm.scratchSpace[scratchIndex] = pgm.stack[top]
+	}
+	return nil, nil
+}
+
+func typeStores(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes) {
+	top := len(pgm.stack) - 1
+	if top >= 0 {
+		for i := range pgm.scratchSpace {
+			if pgm.scratchSpace[i] != pgm.stack[top] {
+				pgm.scratchSpace[i] = StackAny
+			}
+		}
+	}
+	return nil, nil
+}
+
+func typeLoad(pgm *ProgramKnowledge, args []string) (StackTypes, StackTypes) {
+	if len(args) == 0 {
+		return nil, nil
+	}
+	n, err := strconv.ParseUint(args[0], 0, 64)
+	if err != nil {
+		return nil, nil
+	}
+	scratchIndex := int(n)
+	if scratchIndex < 0 || scratchIndex > 255 {
+		return nil, nil
+	}
+	return nil, StackTypes{pgm.scratchSpace[scratchIndex]}
```

Comment:
Maybe we abstract grabbing a single immediate number, so the several checks (arg exists, it parses, it's under 256) can all happen in one function.

Reply:
Good point, done.